tests/routes/contacts.test.js

```diff
@@ -97,6 +106,56 @@ describe('/contacts route', () => {
       // but second call should do nothing
       expect(afterContacts.body).toHaveLength(beforeContacts.body.length);
     });
+
+    test('Connects to CSAM reports', async () => {
+      const updateSpy = jest.spyOn(CSAMReport, 'update');
+
+      // Create CSAM Report
+      const csamReportId1 = 'csam-report-id-1';
+      const csamReportId2 = 'csam-report-id-2';
+
+      const newReport1 = (
+        await CSAMReportController.createCSAMReport(
+          {
+            csamReportId: csamReportId1,
+            twilioWorkerId: workerSid,
+          },
+          accountSid,
+        )
+      ).dataValues;
+
+      const newReport2 = (
+        await CSAMReportController.createCSAMReport(
+          {
+            csamReportId: csamReportId2,
+            twilioWorkerId: workerSid,
+          },
+          accountSid,
+        )
+      ).dataValues;
+
+      // Create contact with above report
+      const response = await request
+        .post(route)
+        .set(headers)
+        .send({ ...contact1, csamReports: [newReport1, newReport2] });
+
+      expect(updateSpy).toHaveBeenCalled();
+
+      const updatedReport1 = await CSAMReportController.getCSAMReport(newReport1.id, accountSid);
+      expect(updatedReport1.contactId).toEqual(response.body.id);
+      expect(updatedReport1.csamReportId).toEqual(csamReportId1);
+
+      const updatedReport2 = await CSAMReportController.getCSAMReport(newReport2.id, accountSid);
+      expect(updatedReport2.contactId).toEqual(response.body.id);
+      expect(updatedReport2.csamReportId).toEqual(csamReportId2);
+
+      // Test the association
+      expect(response.body.csamReports).toHaveLength(2);
+
+      // Remove this contact to not interfere with following tests
+      await Contact.destroy({ where: { id: response.body.id } });
```

Comment:
Should be in a `finally` block so it runs if the rest of the test errors

Reply:
This only makes sure the following tests are not affected, as some tests for contacts relies on the amount of contacts present in the DB. I think wrapping this in a `finally` is better, thanks for the comment, but maybe will be even better to re-do the tests to stop relying on "the amount of records".